packages/react/src/hooks/registration.ts

```diff
@@ -111,21 +109,32 @@ export const useSignUpEmailPassword: SignUpEmailPasswordHook = (
   const stateOptions = c || (typeof a !== 'string' ? a : undefined)
 
   const service = useAuthInterpreter()
-  const isError =
-    !!service.status && service.state.matches({ authentication: { signedOut: 'failed' } })
+  const isError = useSelector(
+    service,
+    (state) => !state.matches({ authentication: { signedOut: 'failed' } })
```

Comment:
Should this really be `!state.matches` and not `state.matches` without the `!` in the beginning?

Reply:
Other than that it looks good! 👍